phone-book/src/app.js

```diff
@@ -0,0 +1,177 @@
+class App {
+  constructor(appContainerId) {
+    this.state = {};
+    this.appContainer = document.getElementById(appContainerId);
+    this.router = new Router(this);
+    this.serverAPI = new ServerAPI();
+    const userPage = new EditUser(this);
+    this.pages = {
+      contacts : {
+        pageObject : new Contacts(this),
+        href : '/contacts.html'
+      },
+      keypad : {
+        pageObject : new Keypad(this),
+        href : '/keypad.html'
+      },
+      editUser : {
+        pageObject : userPage,
+        href : '/edit-contact.html'
+      },
+      user : {
+        pageObject : new User(this),
+        href : '/user.html'
+      },
+      addUser : {
+        pageObject : userPage,
+        href : '/add-user.html'
+      }
+    }
+
+    const appHtml = this.renderAppContainer();
+    this.appContainer.innerHTML = appHtml;
+    this.addRouter();
+
+    this.state.pageName = 'Contacts';
+    this.updateState({activePage : 'contacts'});
+    this.router.gotToPage(this.state,'/index.html');
+    this.changePageToActive();
+  }
+
+  updateState(newState){
+    this.state = {};
+    Object.assign(this.state,newState);
+
+
+  }
+
+  addRouter(){
+    this.appContainer.addEventListener('click', (event) => {
+      event.preventDefault();
+      const container = event.currentTarget;
+      let target = event.target;
+      while (target != container) {
+        if (target.getAttribute("data-page") ) {
+          const newPage = target.getAttribute("data-page");
+          const newUser = target.getAttribute("data-user-id");
+          const newState = { activePage : newPage };
+          if(newUser){
+            newState.userId = newUser;
+          }
+          this.updateState(newState);          
+          this.router.gotToPage(this.state);
+          this.changePageToActive();
+          return;
+        }
+        target = target.parentNode;
+      }
+    });  
+  }
+
+  changePageToActive(){
+    this.updateTitle();
+    this.insertCurrentPage();
+    this.initPageEventListeners();
+  }
+  updateTitle(){    
+    const currentPage = this.getCurrentPage(); 
+    const title = this.appContainer.querySelector('header > div > h2');
+    title.textContent = currentPage.title;
+  }
+
+  getCurrentPage(){
+    const activePage = this.state.activePage;
+    const currentPage = this.pages[activePage].pageObject;  
+    return currentPage;  
+  }
+
+  initPageEventListeners(){
+    const currentPage = this.getCurrentPage(); 
+    currentPage.initEvents();
+  }
+  
+  renderAppContainer(){
+    return `
+    ${this.renderAppHead()}
+    ${this.renderAppPage()}
+    ${this.renderAppFooter()}    
+    `;
+  }
+
+  formatformNumber(number) {
+    if(isNaN(number.charAt(0))){
```

Comment:
why the main module of your app where all initialization happens should know about the internals and business logic of some module such as formatformNumber ? It could never been here. Please move it to required module